plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/policies/FBNConnectionEndpointPolicy.java

```diff
@@ -141,9 +144,20 @@ private static IFigure selectonOverlayFigure(final FBNetworkConnectionLabel labe
 
 		final Label lab = new Label(label.getLabel().getText());
 		lab.setBackgroundColor(org.eclipse.draw2d.ColorConstants.white);
-		lab.setForegroundColor(label.getForegroundColor());
 		lab.setBounds(bounds);
-		lab.setBorder(new LineBorder(label.getForegroundColor(), 2));
+
+		final boolean isErrorLabel = GraphicalAnnotationStyles.findAnnotationBorder(label, AnnotationBorder.class)
+				.isPresent();
+		if (isErrorLabel) {
+			final Border border = GraphicalAnnotationStyles.findAnnotationBorder(label, AnnotationBorder.class)
+					.orElse(null);
+			if (null != border) {
+				lab.setBorder(border);
+			}
```

Comment:
I believe this can now be simplified to:
```suggestion
		final Optional<AnnotationBorder> annotationBorder = GraphicalAnnotationStyles.findAnnotationBorder(label, AnnotationBorder.class);
		if (annotationBorder.isPresent()) {
			lab.setBorder(annotationBorder.get());
```